lib/THC/generic/THCTensorMathScan.cu

```diff
@@ -57,12 +78,22 @@ template<class BinaryFunction>
 void THCTensor_(scanDim)(THCState *state, THCTensor *self_, THCTensor *src,
                          long dimension, real init, BinaryFunction binary_op)
 {
-  THCTensor_(resizeAs)(state, self_, src);
+  // "init" must be the identity element for binary_op
+  int ndim = THCTensor_(nDimension)(state, src);
+  THArgCheck(dimension >= 0 && dimension < ndim, 3, "dimension %d out of range",
+      dimension + TH_INDEX_BASE);
 
+  THCTensor_(resizeAs)(state, self_, src);
   THCTensor *self = THCTensor_(newContiguous)(state, self_);
   src = THCTensor_(newContiguous)(state, src);
 
-  if (dimension == THCTensor_(nDimension)(state, src) - 1) {
+#ifndef THC_REAL_IS_HALF
+  if (ndim == 1) {
```

Comment:
don't we need to check that the tensors are contiguous? Or they are supposed to be contiguous at this point?